x-pack/filebeat/input/azureeventhub/sanitization.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package azureeventhub
+
+import "strings"
+
+type sanitizationFunc func(jsonStr string) []byte
+
+func getSanitizationFuncs() map[string]sanitizationFunc {
+	return map[string]sanitizationFunc{
```

Comment:
Wouldn't a switch/case be more readable here (it would also allow having a default behavior).

Reply:
Should the default behaviour be for all sanitizations to occur for Azure? What's your opinion on this? @zmoog

Reply:
1. The default behavior must be no sanitization; these malformed JSON documents are edge cases.
2. We need to make this option available in the agent template file and on the integration settings page. 

For example, we can turn it on by default for FunctionApp logs but also give the option to turn it off as an escape hatch.

This triggers a question: we will probably release this change with stack version 8.8 and backport it to 8.7.2. How can we make this fix available without raising the minimum required version for the Azure Integration logs to 8.7.2? Today the min required version is `^7.16.0 || ^8.0.0`. 

It's a big bump to address and an edge case.